lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -92,6 +108,16 @@ public function __construct($query, $parserResult, array $queryComponents)
      */
     public function walkSelectStatement(SelectStatement $AST)
     {
+        // In the case of ordering a query by columns from joined tables, we
+        // must add those columns to the select clause of the query BEFORE
+        // the SQL is generated.
+        $this->addMissingItemsFromOrderByToSelect($AST);
+
+        // Remove order by clause from the inner query
+        // It will be re-appended in the outer select generated by this method
+        $orderByClause = $AST->orderByClause;
+        $AST->orderByClause = null;
```

Comment:
essentially an order by clause in a subquery without a limit applied is meaningless - there may be scenarios where ordering by in a subquery can be implemented by a DBA to help with joins on subordinate keys, but the ORM doesn't do that.